anchore_engine/services/catalog/catalog_impl.py

```diff
@@ -2196,32 +2191,85 @@ def _delete_image_for_real(userId, image_record, dbsession, image_ids, image_ful
         "manifest_data",
         "parent_manifest_data",
     ]:
-        archiveId = imageDigest
-        logger.debug("DELETEing image from archive {}/{}".format(bucket, archiveId))
-        rc = obj_store.delete(userId, bucket, archiveId)
+        try:
+            logger.debug("DELETEing image from archive %s/%s", bucket, image_digest)
+            rc = obj_store.delete(account_id, bucket, image_digest)
+        except Exception:
+            logger.exception(
+                "Error deleting image from archive %s/%s", bucket, image_digest
+            )
 
     # digest/tag-based archiveId buckets
     for bucket in ["vulnerability_scan"]:
-        for fulltag in image_fulltags:
-            archiveId = "{}/{}".format(imageDigest, fulltag)
-            logger.debug("DELETEing image from archive {}/{}".format(bucket, archiveId))
-            rc = obj_store.delete(userId, bucket, archiveId)
+        for full_tag in full_tags:
+            archive_id = "{}/{}".format(image_digest, full_tag)
+            try:
+                logger.debug("DELETEing image from archive %s/%s", bucket, archive_id)
+                rc = obj_store.delete(account_id, bucket, archive_id)
+            except Exception:
+                logger.exception(
+                    "Error deleting image from archive %s/%s", bucket, archive_id
+                )
 
-    logger.debug("DELETEing image from policy_engine")
 
+def _delete_image_catalog(account_id, image_digest, image_ids, full_tags, db_session):
+    try:
+        logger.debug("DELETEing image from catalog")
+        rc = db_catalog_image.delete(image_digest, account_id, session=db_session)
+    except Exception:
+        logger.exception("Error deleting image digest %s from catalog", image_digest)
+
+
+def _delete_image_policy_engine(
+    account_id, image_digest, image_ids, full_tags, db_session
+):
     # prepare inputs
+
     try:
-        pe_client = internal_client_for(PolicyEngineClient, userId=userId)
+        pe_client = internal_client_for(PolicyEngineClient, userId=account_id)
         for img_id in set(image_ids):
             logger.debug(
-                "DELETING image from policy engine userId = {} imageId = {}".format(
-                    userId, img_id
-                )
+                "DELETING image from policy engine account_id=%s image_id=%s",
+                account_id,
+                img_id,
             )
-            rc = pe_client.delete_image(user_id=userId, image_id=img_id)
-    except:
-        logger.exception("Failed deleting image from policy engine")
-        raise
+            rc = pe_client.delete_image(user_id=account_id, image_id=img_id)
+    except Exception:
```

Comment:
Exception is too broad this should be ValueError or extending the exceptions.py module to include a new Type that is more specific to this kind of error.